src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWriterAsync.cs

```diff
@@ -599,5 +599,22 @@ private async Task WriteLocalNamespacesAsync(XPathNavigator nsNav)
                 await WriteAttributeStringAsync("xmlns", prefix, XmlReservedNs.NsXmlNs, ns).ConfigureAwait(false);
             }
         }
+
+        public ValueTask DisposeAsync()
+        {
+            if (WriteState != WriteState.Closed)
+            {
+                return DisposeAsyncCore();
+            }
+
+            return default;
+        }
+
+        protected virtual ValueTask DisposeAsyncCore()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
Shouldn't this call to GC.SuppressFinalize be in DisposeAsync()?

Reply:
This is what i understood from API review CC @bartonjs, though i might have called `DisposeAsyncCore();` from `DisposeAsync()` unconditionally, the condition should be here

Reply:
That doesn't make sense to me.  The whole purpose of separating out the DisposeAsyncCore was so that DisposeAsync could call SuppressFinalize.

Reply:
I see, will update